File: Solutions/647. Palindromic Substrings/647. Palindromic Substrings.go
var res int

func countSubstrings(s string) int {
    res=0
    for i:=0;i<len(s);i+=1{
        extendPal(s,i,i)
        extendPal(s,i,i+1)
    }
    return res
}

func extendPal(s string,i,j int){
    for i>=0 && j<len(s) && s[i]==s[j]{
        res,i,j=res+1,i-1,j+1
    }
}

File: Solutions/647. Palindromic Substrings/647. Palindromic Substrings.cpp
class Solution {
public:
    int res=0;
    int countSubstrings(string s) {
        for(int i=0;i<s.size();++i){
            extendPal(s,i,i);
            extendPal(s,i,i+1);
        }
        return res;
    }
    void extendPal(string& s,int i,int j){
        while(i>=0 and j<s.size() and s[i]==s[j]){
            ++res,--i,++j;
        }
    }
};